### DISCUSSION TITLE
Persist favorite/rating status for file system music providers

### DISCUSSION THREAD

**Discussion by madbrain76** [music-assistant/support#2463] [Feature requests and ideas]
Recently, I installed, un-installed and reinstalled MA. I was surprised that the favorite flags on my music library stored on the file system didn't reappear. It turns out the flags are stored in a local MA database, which needs to be backed up.

The user experience is different than in say, the Qobuz provider, where the favorite status is synchronized bo5h ways. That difference was confusing to me.

I'm proposing an enhancement to make this more consistent across providers.
1. One option is to add metarata to the files themselves, assuming the directory/share is writeable, which is currently the case.
2. Since there doesn't seem to be a standard favorite metadata element, perhaps use the more common rating element.  See my separate RFE on rating support at  https://github.com/orgs/music-assistant/discussions/2461 .
3. some file formats do not support all metadata. Sidecar files similar to XMP could be used in this case. The downside is that external applications will not move them as needed.
4. the long-shot option is to use extended file attributes. These should work with all file types. But probably not all file systems, unfortunately.
5. Since I expect not everyone will want this feature, it probably should be opt in or opt out. If the share/FS is read only, it would be disabled. Although at this moment I believe write access is required.
6. When the feature is disabled, there should be a message explaining the treatment of these flags, and the need for backup. The right time to do that is probably when adding the file system provider.

**Comment by MarvinSchenkel** [comment]
Moving music-assistant/backlog#12 here for community feedback. The specific proposal is to optionally mark local tracks as favorites from their rating tags (for example, four stars or more), and write a rating back to the local file when it is favorited in MA. Please upvote and share which rating tags and file-based players you use.
